### PR TITLE
fix(client): Vague Error messages

### LINKED ISSUE #17413: We shouldn't use vague error messages anymore
We're a community of developers, and I think we should surface more of what the error is actually happening in cases where it isn't a likely security risk. If feel like error messages like "oops - something went wrong" are kind of insulting to users' intelligence.

Would anyone be interested in auditing the codebase and going through to make the error messages a bit more specific?

### PR DESCRIPTION
**Pull request by thecodingaviator**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
~- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)~
- [x] My changes do not use shortened URLs or affiliate links.

Closes #17413

### CODE REVIEW

File: api-server/common/models/User-Identity.js [hunk 1]
```diff
@@ -50,9 +50,9 @@ export default function(UserIdent) {
         new Error('invalid or empty email recieved from auth0'),
         {
           message: dedent`
-    Oops... something is not right. We did not find a valid email from your
-    ${provider} account. Please try again with a different provider that has an
-    email available with it.
+          ${provider} did not return a valid email address.
```
Comment (raisedadead) [inline api-server/common/models/User-Identity.js]: Fix the indentation like it was originally.

File: api-server/common/models/User-Identity.js [hunk 2]
```diff
@@ -50,9 +50,9 @@ export default function(UserIdent) {
...
+          Please try again with a different account that has an
```
Comment (raisedadead) [inline api-server/common/models/User-Identity.js]: Fix the indentation like it was originally.

File: api-server/common/models/User-Identity.js [hunk 3]
```diff
@@ -50,9 +50,9 @@ export default function(UserIdent) {
...
+    email associated with it or modify permissions to allow us to view your email.
```
Comment (raisedadead) [inline api-server/common/models/User-Identity.js]:
```suggestion
    email associated with it or update the settings on ${provider}, for us to be able to retrieve your email.
```

File: api-server/server/boot/authentication.js [hunk 4]
```diff
@@ -53,7 +53,7 @@ module.exports = function enableAuthentication(app) {
       if (err) {
         throw wrapHandledError(new Error('could not destroy session'), {
           type: 'info',
-          message: 'Oops, something is not right.',
+          message: 'We could not log you out, please try logging out again.',
```
Comment (raisedadead) [inline api-server/server/boot/authentication.js]:
```suggestion
          message: 'We could not log you out, please try again in a moment.',
```

Comment (raisedadead) [reply api-server/server/boot/authentication.js]: Give them the impression, that something is delayed on our side.

File: api-server/server/boot/user.js [hunk 5]
```diff
@@ -202,10 +202,15 @@ function createPostReportUserProfile(app) {
...
+    if (!username) {
       return res.json({
         type: 'danger',
-        message: 'Oops, something is not right please re-check your submission.'
+        message: 'Please enter a valid username.'
+      });
+    } else if (!report || report === '') {
+      return res.json({
+        type: 'danger',
+        message: 'Your report cannot be empty. Please add information.'
```
Comment (raisedadead) [inline api-server/server/boot/user.js]:
```suggestion
        message: 'Your report cannot be empty. Please add details for us to be able to address your concern.'
```

File: api-server/server/boot/user.js [hunk 6]
```diff
@@ -202,10 +202,15 @@ function createPostReportUserProfile(app) {
...
     log(report);
 
-    if (!username || !report || report === '') {
+    if (!username) {
       return res.json({
         type: 'danger',
-        message: 'Oops, something is not right please re-check your submission.'
+        message: 'Please enter a valid username.'
+      });
+    } else if (!report || report === '') {
```
Comment (raisedadead) [inline api-server/server/boot/user.js]: This conditional is not valid. What happens when `username`, `report` both are null/undefined?

File: api-server/server/middlewares/error-handlers.js [hunk 7]
```diff
@@ -26,7 +26,7 @@ export default function prodErrorHandler() {
 
     const redirectTo = handled.redirectTo || `${homeLocation}/`;
     const message =
-      handled.message || 'Oops! Something went wrong. Please try again later';
+      handled.message || 'Oops! Something went wrong. Please try again';
```
Comment (raisedadead) [inline api-server/server/middlewares/error-handlers.js]:
```suggestion
      handled.message || 'Oops! Something went wrong. Please try again in a moment.';
```

**Review by raisedadead — CHANGES_REQUESTED** [review]

File: api-server/common/models/User-Identity.js (hunk 3 above)
Comment (thecodingaviator) [reply api-server/common/models/User-Identity.js]: How about `update your settings on ${provider}`?

**Review by raisedadead — CHANGES_REQUESTED** [review]
This entire block is anti-pattern. Why? Repetitive, Complex and difficult to read.

```js
    if (!username && (!report || report === '')) {
      return res.json({
      return res.json({
        type: 'danger',
        type: 'danger',
        message: 'Oops, something is not right please re-check your submission.'
        message: 'Please enter a valid username and add details for us to be able to address your concern.'
      });
    } else if ((!report || report === '') && username) {
      return res.json({
        type: 'danger',
        message: 'Your report cannot be empty. Please add details for us to be able to address your concern.'
      });
    } else if (!username) {
      return res.json({
        type: 'danger',
        message: 'Please enter a valid username.'
      });
      });
    }
    }
```
You should re-factor this to be a simple return early pattern.

**Comment by raisedadead** [comment]
Finally I have noticed that you depend on the CI for while fixing up PRs, IMHO you should test things locally before pushing to ease the notifications barrage it triggers for everyone subscribed to the thread.

**Review by raisedadead — APPROVED** [review]

**Comment by raisedadead** [comment]
You should probably have different returns for each of those last validations on username and report, returning early. Anyways I'll merge this.

**Pull request merged**